drivers/net/mlx5/mlx5_vdpa.c

```diff
@@ -147,14 +160,92 @@ mlx5_vdpa_query_virtio_caps(struct vdpa_priv *priv)
 	 * For now only set protocol features support
 	 */
 	priv->caps.max_num_virtqs = MLX5_VDPA_SW_MAX_VIRTQS_SUPPORTED;
-	priv->caps.virtio_net_features = (1ULL << VHOST_USER_F_PROTOCOL_FEATURES);
+	/*
+	 * TODO (shahafs): Take from QUERY HCA CAP Device Emulation Capabilities.
+	 * For now only set protocol features support
+	 */
+	priv->caps.notify_stride = MLX5_VDPA_SW_NOTIFY_STRIDE;
+	priv->caps.virtio_net_features = MLX5_VDPA_FEATURES;
+	priv->caps.virtio_protocol_features = MLX5_VDPA_PROTOCOL_FEATURES;
 	DRV_LOG(DEBUG, "Virtio Caps:");
 	DRV_LOG(DEBUG, "	dump_mkey=0x%x ", priv->caps.dump_mkey);
 	DRV_LOG(DEBUG, "	max_num_virtqs=0x%x ", priv->caps.max_num_virtqs);
 	DRV_LOG(DEBUG, "	features_bits=0x%" PRIx64, priv->caps.virtio_net_features);
 	return 0;
 }
 
+#define MLX5_IB_MMAP_CMD_SHIFT 8
+#define MLX5_IB_MMAP_CMD_MASK ((1 << MLX5_IB_MMAP_CMD_SHIFT) - 1)
```

Comment:
Hi @shahafsh 
should be ((1 << MLX5_IB_CMD_SIZE) - 1) right ? 
It happens to be they are both 8 bits but it's confusing.

Reply:
No.

the layout of the mmap offset is:
page_offset[15:8] | mmap_command[7:0] | page_offset[7:0]

or w/ macro write
page_offset[15:MLX5_IB_MMAP_CMD_SHIFT] | mmap_command[(MLX5_IB_CMD_SIZE-1):0] | page_offset[(MLX5_IB_MMAP_CMD_SHIFT-1):0]

i think the name MLX5_IB_MMAP_CMD_MASK is confusing. will change it.